tests/test_AllocateSizeMoreThanClusterSize.py

```diff
@@ -0,0 +1,240 @@
+"""
+Automates the following test
+
+OCS-269 - FT-OCP-Create-PV-AllocateSizeMoreThanClusterSize
+Verify a PVC creation by allocating more storage
+than what is available in Ceph
+
+https://polarion.engineering.redhat.com/polarion/#/project/
+OpenShiftContainerStorage/workitem?id=OCS-269
+
+"""
+
+import logging
+import yaml
+import os
+from utility.utils import run_cmd
+import json
+from ocs import ocp
+from openshift.dynamic import DynamicClient
+from utility import utils, templating
+from kubernetes import config
+from ocsci.enums import StatusOfTest
+import time
+
+k8s_client = config.new_client_from_config()
+dyn_client = DynamicClient(k8s_client)
+log = logging.getLogger(__name__)
+
+PVC_YAML = os.path.join("templates/ocs-deployment",
+                        "PersistentVolumeClaim.yaml")
+POD_YAML = os.path.join("templates/ocs-deployment",
+                        "Pod.yaml")
+SC_YAML = os.path.join("templates/ocs-deployment",
+                       "StorageClass.yaml")
+TEMP_PVC_YAML_FILE = '/tmp/pvc_test.yaml'
+TEMP_POD_YAML_FILE = '/tmp/pod_test.yaml'
+TEMP_SC_YAML_FILE = '/tmp/sc_test.yaml'
+
+POD = ocp.OCP(kind='Pod')
+PVC = ocp.OCP(kind='PersistentVolumeClaim')
+PV = ocp.OCP(kind='PersistentVolume')
+SC = ocp.OCP(kind='StorageClass')
+NAMESPACE = ocp.OCP(kind='Project')
+
+
+def ceph_storage_capacity(ceph_tool):
+    '''
+    Returns the total capacity of the ceph cluster
+    in openshift-storage project
+    '''
+
+    # TODO: use an exec function to run ceph status
+    pods = run_cmd(f'oc rsh -n openshift-storage {ceph_tool} \
+    ceph status --format json')
+    pods_json = json.loads(pods)
+    avail_cap = pods_json['pgmap']['bytes_total']
+    avail_cap_gb = avail_cap / (1024 * 1024 * 1024)
+    return avail_cap_gb
+
+
+def find_cephtool():
+    '''
+    Returns rook-ceph-tools pod from the list of pods in
+    openshift-storage project
+    '''
+
+    v1_pods = dyn_client.resources.get(api_version='v1', kind='Pod')
+    out = v1_pods.get(
+        namespace='openshift-storage', label_selector='app=rook-ceph-tools'
+    )
+    return out['items'][0]['metadata']['name']
+
+
+def create_namespace(**kwargs):
+    '''
+    Creates a project if it is not already available
+    '''
+    project_name = (kwargs['project_name'])
+    project_get = NAMESPACE.get()
+    namespaces = []
+    for i in range(len(project_get['items'])):
+        namespaces.append(project_get['items'][i]['metadata']['name'])
+    log.info(f'checking id project {project_name} already exists')
+    if project_name in namespaces:
+        log.info(
+            f'project {project_name} exists, using the existing namespace'
+        )
+        return True
+    else:
+        log.info(f'creating a new project {project_name}')
+        return run_cmd(f'oc new-project {project_name}')
```

Comment:
I don't think there's a command like that 'new-project'
it should be oc create new namespace {project_name}

Reply:
@RazTamir I've always used on new-project command. please refer - https://docs.openshift.com/container-platform/4.1/web-console/working-with-projects.html#creating-a-project-using-the-CLI_projects

Is there any other reason we should use 'oc create namespace {project_name}' command?

Reply:
Yeah thanks for the link.
I tried to use it and for some reason failed.
Now I see it creates the new namespace and also changing to it.

My only concern is how we can mimic this in our cli?
we'll need to add new-project as CLI operation

Reply:
> Yeah thanks for the link.
> I tried to use it and for some reason failed.
> Now I see it creates the new namespace and also changing to it.
> 
> My only concern is how we can mimic this in our cli?
> we'll need to add new-project as CLI operation

I did not quite get this. can you elaborate, please?

Reply:
@keesturam @RazTamir project creation using CLI: oc new-project <project_name>
We can enhance exec_oc_cmd() since create() adds 'create' to the command

Reply:
from 'oc new-project --help':
If your administrator allows self-service, this command will create a new project for you and assign you as the project admin.
 After your project is created it will become the default project in your config.

I'm not sure what are the implications of such thing that the default will change.  how this will affect other tests

Reply:
https://github.com/red-hat-storage/ocs-ci/issues/105

Reply:
> from 'oc new-project --help':
> If your administrator allows self-service, this command will create a new project for you and assign you as the project admin.
> After your project is created it will become the default project in your config.
> 
> I'm not sure what are the implications of such thing that the default will change. how this will affect other tests

Shouldn't the commands we send via our tests always have '-n <namespace>' argument?

Reply:
> @keesturam @RazTamir project creation using CLI: oc new-project <project_name>
> We can create such a new method to create a new project since create() adds 'create' to the command

Yes. I had thought about it too while writing this function. let's do it.